p2panda-net/src/sync/mod.rs

```diff
@@ -36,7 +36,7 @@ pub async fn initiate_sync<S: AsyncWrite + Send + Unpin, R: AsyncRead + Send + U
 
     // Set up a channel for receiving new application messages.
     let (tx, mut rx) = mpsc::channel(128);
-    let mut sink = PollSender::new(tx).sink_map_err(|e| SyncError::Protocol(e.to_string()));
+    let mut sink = PollSender::new(tx).sink_map_err(|e| SyncError::Critical(e.to_string()));
```

Comment:
If I understand this correctly this is mapping _all_ error types coming from the regarding `Sync` implementation and converting them to one type (previously `Protocol`)?

That would defy the whole reason of having custom error types in sync. Do I missunderstand this code or should we look for a way of passing the actual sync error type further?

Reply:
Cool, I think this can be looked at in a separate PR as part of the "improved error handling in the sync manager" task